zerver/lib/templates.py

```diff
@@ -82,7 +82,7 @@ def display_list(values: List[str], display_limit: int) -> str:
 @items_tuple_to_dict
 @register.filter(name="render_markdown_path", is_safe=True)
 def render_markdown_path(
-    markdown_file_path: str, context: Mapping[str, Any] = {}, pure_markdown: bool = False
+    markdown_file_path: str, context: Dict[str, Any] = {}, pure_markdown: bool = False
```

Comment:
This falls into the [mutable default argument trap](https://zulip.readthedocs.io/en/latest/testing/mypy.html#read-only-types). (And since the original code was ill-typed, it already did.) Django actually does [mutate](https://github.com/django/django/blob/4.0.6/django/template/backends/jinja2.py) the dict. So we need to do the `context: Optional[Dict[str, Any]] = None` and `if context is None: context = {}` dance.

(I broke this in #15349—oops.)